Common/src/gRPC/ExpressionExt.cs

```diff
@@ -0,0 +1,92 @@
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Linq.Expressions;
+
+namespace ArmoniK.Core.Common.gRPC;
+
+public static class ExpressionExt
+{
+  /// <summary>
+  ///   Combines two predicate expressions using a logical AND condition
+  /// </summary>
+  /// <typeparam name="T"> The type of the input parameter the predicate expressions are evaluated on </typeparam>
+  /// <param name="expr1"> The first predicate expression to combine </param>
+  /// <param name="expr2"> The second predicate expression to combine </param>
+  /// <returns> A new predicate expression that represents the logical AND of the two expressions </returns>
+  public static Expression<Func<T, bool>> ExpressionAnd<T>(this Expression<Func<T, bool>> expr1,
+                                                           Expression<Func<T, bool>>      expr2)
+  {
+    var parameter = Expression.Parameter(typeof(T));
+
+    var leftVisitor = new ReplaceExpressionVisitor(expr1.Parameters[0],
+                                                   parameter);
+    var left = leftVisitor.Visit(expr1.Body);
+
+    var rightVisitor = new ReplaceExpressionVisitor(expr2.Parameters[0],
+                                                    parameter);
+    var right = rightVisitor.Visit(expr2.Body);
+
+    return Expression.Lambda<Func<T, bool>>(Expression.AndAlso(left,
+                                                               right),
+                                            parameter);
+  }
+
+  /// <summary>
+  ///   Combines two predicate expressions using a logical OR condition
+  /// </summary>
+  /// <typeparam name="T"> The type of the input parameter the predicate expressions are evaluated on </typeparam>
+  /// <param name="expr1"> The first predicate expression to combine </param>
+  /// <param name="expr2"> The second predicate expression to combine </param>
+  /// <returns> A new predicate expression that represents the logical OR of the two expressions </returns>
+  public static Expression<Func<T, bool>> ExpressionOr<T>(this Expression<Func<T, bool>> expr1,
+                                                          Expression<Func<T, bool>>      expr2)
+  {
+    var parameter = Expression.Parameter(typeof(T));
+
+    var leftVisitor = new ReplaceExpressionVisitor(expr1.Parameters[0],
+                                                   parameter);
+    var left = leftVisitor.Visit(expr1.Body);
+
+    var rightVisitor = new ReplaceExpressionVisitor(expr2.Parameters[0],
+                                                    parameter);
+    var right = rightVisitor.Visit(expr2.Body);
+
+    return Expression.Lambda<Func<T, bool>>(Expression.OrElse(left,
+                                                              right),
+                                            parameter);
+  }
+
+  private class ReplaceExpressionVisitor : ExpressionVisitor
```

Comment:
Pretty clean solution !
I'd use Expression.MakeBinary to factorize code and make the generic function private.